### PR TITLE
Feat(eos_cli_config_gen): Add ISIS authentication and hello padding

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Add ISIS authentication and hello padding

## Related Issue(s)

Needed for PR #1418 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

On `ethernet_interfaces` and `port_channel_interfaces`:
```yaml
    isis_hello_padding: < true | false >
    isis_authentication_mode: < text | md5 >
    isis_authentication_key: < encrypted password >
```

## How to test

Molecule

### User Checklist

- N/A

### Repository Checklist

- [X] My code has been rebased from devel before I start
- [X] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [X] My change requires a change to the documentation and documentation have been updated accordingly.
- [X] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by emilarista — APPROVED** [review]
LGTM!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ethernet-interfaces.j2 [hunk 1]
```diff
@@ -299,7 +299,9 @@
                ethernet_interfaces[ethernet_interface].isis_metric is arista.avd.defined or
                ethernet_interfaces[ethernet_interface].isis_circuit_type is arista.avd.defined or
                ethernet_interfaces[ethernet_interface].isis_network_point_to_point is arista.avd.defined or
-               ethernet_interfaces[ethernet_interface].isis_passive is arista.avd.defined %}
+               ethernet_interfaces[ethernet_interface].isis_passive is arista.avd.defined or
+               ethernet_interfaces[ethernet_interface].isis_hello_padding is arista.avd.defined or
+               pethernet_interfaces[ethernet_interface].isis_authentication_mode is arista.avd.defined %}
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ethernet-interfaces.j2]: typo! ```pethernet```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ethernet-interfaces.j2 [hunk 2]
```diff
@@ -327,29 +331,32 @@
...
 {%                 endif %}
 {%             else %}
 {%                 if ethernet_interface in ethernet_interfaces_isis %}
 {%                     set channel_group = ethernet_interfaces[ethernet_interface].channel_group.id | arista.avd.default("-") %}
 {%                     set isis_instance = ethernet_interfaces[ethernet_interface].isis_enable | arista.avd.default("-") %}
 {%                     set isis_metric = ethernet_interfaces[ethernet_interface].isis_metric | arista.avd.default("-") %}
 {%                     set isis_circuit_type = ethernet_interfaces[ethernet_interface].isis_circuit_type | arista.avd.default("-") %}
-{%                     if ethernet_interfaces[ethernet_interface].isis_network_point_to_point is arista.avd.defined(true) %}
+{%                     set isis_hello_padding = ethernet_interfaces[ethernet_interface].isis_hello_padding | arista.avd.default("-") %}
+{%                     set isis_authentication_mode = ethernet_interfaces[ethernet_interface].isis_authentication_mode | arista.avd.default("-") %}{%                     if ethernet_interfaces[ethernet_interface].isis_network_point_to_point is arista.avd.defined(true) %}
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ethernet-interfaces.j2]: Can you move the if check to a new line.

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ethernet-interfaces.j2]: Thank you. Fixed

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ethernet-interfaces.j2 (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ethernet-interfaces.j2]: Thank you. Fixed

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 3]
```diff
@@ -899,6 +899,9 @@ ethernet_interfaces:
     isis_metric: < integer >
     isis_network_point_to_point: < boolean >
     isis_circuit_type: < level-1-2 | level-1 | level-2 >
+    isis_hello_padding: < true | false >
+    isis_authentication_mode: < text | md5 >
+    isis_authentication_key: < encrypted password >
```
Comment (ankudinov) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: replace with < type-7 encrypted password > or add a comment

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/port-channel-interfaces.yml [hunk 4]
```diff
@@ -377,6 +389,12 @@ ethernet_interfaces:
       id: 109
       mode: active
 
+  Ethernet10/10:
```
Comment (ankudinov) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/port-channel-interfaces.yml:392]: is it `10/10` intentionally?

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 5]
```diff
@@ -1195,6 +1198,9 @@ port_channel_interfaces:
...
```
Comment (ankudinov) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: replace with < type-7 encrypted password > or add a comment

**Review by ankudinov — COMMENTED** [review]
added a few minor comments

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/port-channel-interfaces.yml (hunk 4 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/port-channel-interfaces.yml:392]: all these interface numbers are just there to show that "inherited from port-channel" feature. Yes it is intentional.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 3 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Good point. Will fix

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Fixed

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 5 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Fixed

**Review by tgodaA — APPROVED** [review]
LGTM!

**Review by ankudinov — COMMENTED** [review]
LGTM

**Pull request merged**